module/core/src/Application/DependencyInjection/CompilerPass/InstallerInterfaceCompilerPass.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+/**
+ * Copyright © Bold Brand Commerce Sp. z o.o. All rights reserved.
+ * See LICENSE.txt for license details.
+ */
+
+declare(strict_types = 1);
+
+namespace Ergonode\Core\Application\DependencyInjection\CompilerPass;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Reference;
+use Ergonode\Core\Application\Installer\InstallerProvider;
+
+/**
+ */
+class InstallerInterfaceCompilerPass implements CompilerPassInterface
+{
+    public const TAG = 'core.application.installer_interface';
+
+    /**
+     * @param ContainerBuilder $container
+     */
+    public function process(ContainerBuilder $container): void
+    {
+        if ($container->has(InstallerProvider::class)) {
```

Comment:
You could use just configuration annotation rather than compiler pass (with `!tagged_iterator`)